networkx/algorithms/similarity.py

```diff
@@ -1642,6 +1643,170 @@ def panther_similarity(
     return top_k_with_val
 
 
+@nx._dispatchable(edge_attrs="weight")
+def panther_vector_similarity(
+    G, source, D=50, k=5, path_length=5, c=0.5, delta=0.1, eps=None, weight="weight"
+):
+    r"""Returns the Panther vector similarity of nodes in `G`.
+
+     This approach to similarity is often called "Panther++" and the notion of
+     similarity referred to here is low "graph edit distance". Panther is a
+     similarity metric that says "two objects are considered to be similar
+     if their structures are similar" [1]_. In this case, if their graph edit
+     distances are small.
+
+    Parameters
+    ----------
+    G : NetworkX graph
+        A NetworkX graph
+    source : node
+        Source node for whom to find the top ``k`` similar other nodes
+    D : int
+        The number of similarity scores to use (in descending order)
+        for each feature vector. Defaults to 50 per [1]_
+    k : int
+        The number of most similar nodes to return
+    path_length : int
+        How long the randomly generated paths should be (``T`` in [1]_)
+    c : float
+        A universal positive constant. Defaults to 0.5 per [1]_
+    delta : float
+        The probability that $S$ is not an epsilon-approximation to (R, phi)
+    eps : float
+        The error bound. Per [1]_, a good value is ``sqrt(1/|E|)``. Therefore,
+        if no value is provided, the recommended computed value will be used.
+    weight : string or None, optional (default="weight")
+        The name of an edge attribute that holds the numerical value
+        used as a weight. If None then each edge has weight 1.
+
+    Returns
+    -------
+    similarity : dict
+        Dict of nodes to similarity scores (as floats).
+        Note: the self-similarity (i.e., `node`) is not included in the dict.
+
+    Examples
+    --------
+
+        >>> import networkx as nx
+        >>> G = nx.star_graph(10)
+        >>> sim = nx.panther_vector_similarity(G, 0)
+
+    References
+    ----------
+    .. [1] Zhang, J., Tang, J., Ma, C., Tong, H., Jing, Y., & Li, J.
+           Panther: Fast top-k similarity search on large networks.
+           In Proceedings of the ACM SIGKDD International Conference
+           on Knowledge Discovery and Data Mining (Vol. 2015-August, pp. 1445–1454).
+           Association for Computing Machinery. https://doi.org/10.1145/2783258.2783267.
+    """
+    import numpy as np
+    from scipy.spatial import cKDTree as KDTree
+
+    if source not in G:
+        raise nx.NodeNotFound(f"Source node {source} not in G")
+
+    isolates = set(nx.isolates(G))
+
+    if source in isolates:
+        raise nx.NetworkXUnfeasible(
+            f"Panther similarity is not defined for the isolated source node {source}."
+        )
+
+    num_nodes = G.number_of_nodes()
+    if num_nodes < k:
+        warnings.warn(
+            f"Number of nodes is {num_nodes}, but requested k is {k}. "
+            "Setting k to number of nodes."
+        )
+        k = num_nodes
```

Comment:
I'd prefer to raise when k > len(G) instead of trying to guess what the user wanted.
Same is true for D and maybe others.